### PR TITLE
Use pub.dev in writing

### PR DESCRIPTION
**Pull request by sigurdm**
Change all mentions of the pub.dartlang.org website to pub.dev in the ui.

Fixes: https://github.com/dart-lang/pub/issues/3519

### CODE REVIEW

File: lib/src/command/lish.dart [hunk 1]
```diff
@@ -154,9 +154,9 @@ class LishCommand extends PubCommand {
   Future<void> _publish(List<int> packageBytes) async {
     try {
       final officialPubServers = {
-        'https://pub.dartlang.org',
-        // [validateAndNormalizeHostedUrl] normalizes https://pub.dev to
-        // https://pub.dartlang.org, so we don't need to do allow that here.
+        'https://dev',
```
Comment (timsneath) [inline lib/src/command/lish.dart:157]: Should this be pub.dev?

Comment (sigurdm) [reply lib/src/command/lish.dart:157]: Thanks for catching this!

**Comment by timsneath** [comment]
Is this ready to be merged, @sigurdm?

**Comment by sigurdm** [comment]
Unfortunately I think we are going to backpedal a bit here.

When trying it out some unpleasing behaviour became apparent when using an older pub with a pubspec.lock created after this patch: Pub (and dart run) would look for cached entries based on the server in the pubspec.lock (instead of the package_config.json as one could argue it should), and crash when it could not find them (fixed in https://github.com/dart-lang/pub/pull/3533)

I am working on a reduced version that only updates the writing, but still uses pub.dartlang.org for all actual operations.

**Comment by timsneath** [comment]
Oh, for sure. So long as we can fix the UI text itself, it's fine if we still use pub.dartlang.org behind the scenes. Thanks!

**Review by jonasfj — APPROVED** [review]

**Pull request merged**